src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/BaseRecalibratorDataflow2.java

```diff
@@ -0,0 +1,282 @@
+    protected void setupPipeline(Pipeline pipeline) {
+        try {
+            bunny.start("BaseRecalibratorDataflow");
+
+            if (null==baseRecalibrationKnownVariants || baseRecalibrationKnownVariants.size()==0) {
+                throw new UserException.CommandLineException(NO_DBSNP_EXCEPTION);
+            }
+
+            // TODO: once we move to our own thing, make RECAL_TABLE_FILE optional
+            saveTextualTables = (null != BRAC.RAC.RECAL_TABLE_FILE);
+
+            if (!RefAPISource.isApiSourceUrl(referenceURL)) {
+                throw new UserException.CouldNotReadInputFile("Only API reference names are supported for now (start with " + RefAPISource.URL_PREFIX + ")");
+            }
+            referenceID = RefAPISource.getReferenceSetID(referenceURL);
+
+            if (BRAC.readArguments.getReadFilesNames().size()!=1) {
+                throw new UserException("Sorry, we only support a single input file for now.");
+            }
+            String bam = BRAC.readArguments.getReadFilesNames().get(0);
+
+            // Load the input bam
+            final ReadsDataflowSource readsDataflowSource = new ReadsDataflowSource(bam, pipeline);
+            readsHeader = readsDataflowSource.getHeader();
+            final SAMSequenceDictionary readsDictionary = readsHeader.getSequenceDictionary();
+            final List<SimpleInterval> intervals = BRAC.intervalArgumentCollection.intervalsSpecified() ? BRAC.intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary())
+                : IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
+            final PCollectionView<SAMFileHeader> headerSingleton = ReadsDataflowSource.getHeaderView(pipeline, readsHeader);
+            final PCollection<GATKRead> filteredReads = readsDataflowSource.getReadPCollection(intervals)
+                .apply(new DataflowReadFilter(BaseRecalibratorDataflow2.readFilter(readsHeader), readsHeader));
+
+            bunny.stepEnd("set up bam input");
+
+            // GATKRead -> how many reference bases we need for the BAQ computation
+            final SerializableFunction<GATKRead, SimpleInterval> baqWindowFunction = new SerializableFunction<GATKRead, SimpleInterval>() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public SimpleInterval apply( GATKRead read ) {
+                    return BAQ.getReferenceWindowForRead(read, BAQ.DEFAULT_BANDWIDTH, BAQ.DEFAULT_INCLUDE_CLIPPED_BASES);
+                }
+            };
+
+
+            // Load the Variants and the Reference
+            final VariantsDataflowSource variantsDataflowSource = new VariantsDataflowSource(baseRecalibrationKnownVariants, pipeline);
+
+            final KV<SAMSequenceDictionary, Map<String, String>> refDicAndNameToID = RefAPISource.getInstance().getReferenceSequenceDictionaryAndBuildReferenceNameToIdTable(pipeline.getOptions(), referenceID, readsDictionary);
```

Comment:
Well the reason they're unified is that they both grab the same underlying data very slowly and then munge it in slightly different ways. Keeping them separate would halve the performance. I have fixed the underlying data-grabbing performance in a separate PR, but it still seems silly to grab the exact same data twice. Let me try something different, let me know if you like it better.